lib/cdo/contact_rollups.rb

```diff
@@ -57,9 +58,10 @@ class ContactRollups
   def self.build_contact_rollups
     start = Time.now
 
-    ActiveRecord::Base.connection.execute "SET SQL_SAFE_UPDATES = 0"
-    # set READ UNCOMMITTED transaction isolation level to avoid taking locks on tables we are reading from during
-    # what can be multi-minute operations
+    PEGASUS_REPORTING_DB_WRITER.run "SET SQL_SAFE_UPDATES = 0"
+    # set READ UNCOMMITTED transaction isolation level on both read connections to avoid taking locks
+    # on tables we are reading from during what can be multi-minute operations
+    DASHBOARD_REPORTING_DB_READER.run "SET SESSION TRANSACTION ISOLATION LEVEL READ UNCOMMITTED"
```

Comment:
Does the `ActiveRecord::Base.connection` below not accomplish this for the `DASHBOARD_REPORTING_DB_READER`? My intuition is that you would be adding this for `PEGASUS_REPORTING_DB_READER`.